subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/repository/ResolvingWithPluginManagementSpec.groovy

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.repository
+
+import org.gradle.integtests.fixtures.AbstractDependencyResolutionTest
+import org.gradle.test.fixtures.file.LeaksFileHandles
+import org.gradle.test.fixtures.plugin.PluginBuilder
+
+@LeaksFileHandles
+class ResolvingWithPluginManagementSpec extends AbstractDependencyResolutionTest {
```

Comment:
we should have some test coverage for:

- what happens when there are multiple rules attached to the plugin resolution strategy
- what happens when a rule throws an exception
- what happens when build logic tries to mutate the plugin management state after the settings have been evaluated (eg root project doesn't use `plugin { }` block tries to change settings prior to child project that does use `plugins { }` block, or init script that retains reference to plugin management object and tried to mutate later).